multi-agent/lib/mas/blueprints/exceptions.py

```
@@ -1,43 +1,128 @@
+    Raised when attempting to insert a version snapshot that already exists.
+
+    The ``(blueprint_id, version)`` compound key is unique; this exception
+    is the domain-layer translation of the infrastructure-level
+    ``DuplicateKeyError`` from MongoDB.
+    """
+
+    def __init__(self, blueprint_id: str, version: int) -> None:
+        self.blueprint_id = blueprint_id
+        self.version = version
+        super().__init__(
+            f"Snapshot already exists for blueprint {blueprint_id!r} " f"at version {version}"
```

Ruff (0.15.17)

[warning] Implicitly concatenated string literals on one line

Combine string literals

(ISC001)